### DISCUSSION TITLE
Static assets upload returns 404 in production

### DISCUSSION THREAD

**Discussion by arso1er** [vercel/next.js#14121] [Help] [answered]
# Bug report

## Describe the bug
In production, when I upload an image to the public folder, next.js returns a 404 error when trying to view the image. I have to restart the server for next.js to respond with a 200 to the request.
In dev mode, it works without needing to restart or recompile.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Clone the repository at [https://github.com/arso1er/next.js-static-404](url)
2. Install the dependencies with `yarn install`
3. Run `yarn build` then `yarn start`
4. Open `localhost:3000` in your browser, and choose an image to upload
5. The image will be uploaded to `public/assets/img/users` folder, but it won't be displayed next to the image picker (404 error). 

## Expected behavior
Expected the same behavior as dev mode ie the image should be displayed after upload and without the need to restart the server.

## System information

- OS: Windows 10 (WSL)
- Browser: Chrome v83.0.4103.97
- Version of Next.js: 9.4.4
- Version of Node.js: 12.16.3

**Comment by stephencweiss** [comment]
Fwiw - also running into this after following along with the demo. Assets seem to be skipped within the build step. Is there a configuration that needs to be set?

**Comment by arso1er** [comment, marked as answer]
Hi, didn't know this was moved here.

I solved this by using express to serve my static files
`server.use(express.static(path.join(__dirname, 'public')));`

**Reply by vrinch** [reply]
> Hi, didn't know this was moved here.
> 
> I solved this by using express to serve my static files
> `server.use(express.static(path.join(__dirname, 'public')));`

Hello @arso1er 

How did you use this code in your project and what document

**Reply by HamzaMihfad** [reply]
can you please give us more details, how you implimented this?

**Reply by arso1er** [reply]
Hi,

Sorry for my late reply. You can take a look at this repo: [https://github.com/arso1er/next.js-with-express](https://github.com/arso1er/next.js-with-express).

It's a clone of the `custom-server-express` example from next.js repo. All I did was make express serve static files by adding `lines 1 and 14` to **server.js**.

Hope it helps.

**Reply by Lukortech** [reply]
> Hi,
> 
> Sorry for my late reply. You can take a look at this repo: https://github.com/arso1er/next.js-with-express.
> 
> It's a clone of the `custom-server-express` example from next.js repo. All I did was make express serve static files by adding `lines 1 and 14` to **server.js**.
> 
> Hope it helps.

Seems like a dead end for me.

**Comment by kmvan** [comment]
I have to restart node server after uploaded image via pages api.

**Reply by deshario** [reply]
Is there any way to use uploaded image without restarting the server in nextjs 11.1.0+

**Reply by yahya-aghdam** [reply]
> I have to restart node server after uploaded image via pages api.

Yes I have the same issue.

Can you find a way?

**Reply by deshario** [reply]
> > I have to restart node server after uploaded image via pages api.
> 
> Yes I have the same issue.
> 
> Can you find a way?

You need to upload images outside of /public and serve it via express, I think it should work

**Reply by yahya-aghdam** [reply]
> > > I have to restart node server after uploaded image via pages api.
> > 
> > 
> > Yes I have the same issue.
> > Can you find a way?
> 
> You need to upload images outside of /public and serve it via express, I think it should work

Thanks for reply.
I created an api with next.js default server and it's worked finally!

**Reply by massatron1** [reply]
Thanks for reply.

**Comment by neelhabib** [comment]
```
const express = require("express");
const next = require("next");
const path = require("path");
const port = parseInt(process.env.PORT, 10) || 3000;
const dev = process.env.NODE_ENV !== "production";
const app = next({ dev });
const handle = app.getRequestHandler();

app.prepare().then(() => {
  const server = express();

  server.use(express.static("public")); // this is what you need.

  server.all("*", (req, res) => {
    return handle(req, res);
  });

  server.listen(port, (err) => {
    if (err) throw err;
    console.log(`> Ready on http://localhost:${port}`);
  });
});
```


the above code works for me.

**Comment by FullstackWEB-developer** [comment]
I got same issue in Nextjs 14 app router.
How can I sove this?
